### PR TITLE
Dockerising action

### PR DESCRIPTION
**Pull request by asangas**
Using Docker for running the action. This is mainly to get around with incompatibility issues with Python setup on Amazon Linux. 

(We may need to revise whether we need to stick with AL for runners later, which doesn't seem to be a supported architecture)

### CODE REVIEW

**Review by guyjacksongit — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+COPY Pipfile* /tmp/
+WORKDIR /tmp
+RUN pipenv install --system --ignore-pipfile
+
+USER appuser
+WORKDIR /app
+
+COPY pat_helper.py /app/
+
+ENTRYPOINT ["python", "/app/pat_helper.py"]
```
Comment (ps-jay) [inline Dockerfile]: missing an EOL (new-line) here

File: pat_helper.py [hunk 2]
```diff
@@ -194,8 +199,9 @@ def revoke_pat():
     PAT = 'revoked'
 
 def print_outputs():
-    print(f"::set-output name=pat::${PAT}")
-    print(f"::set-output name=pat_id::${PAT_ID}")
+    print(f"::set-output name=username::{USERNAME}")
```
Comment (ps-jay) [inline pat_helper.py:202]: needed for revoking I guess ?

**Review by ps-jay — APPROVED** [review]

File: pat_helper.py (hunk 2 above)
Comment (asangas) [reply pat_helper.py:202]: Yeah, needed for revoking. We have defined the PAT_ID as an output in the metadata, but haven't set it in the script. 

Also removing the `$` character which messed up the outputs.

File: Dockerfile (hunk 1 above)
Comment (asangas) [reply Dockerfile]: Thanks. Fixed

**Pull request merged**